test/martha/martha_v3.test.js

```diff
@@ -424,412 +382,332 @@ test.serial('martha_v3 returns an error when an invalid field is requested', asy
+
     t.is(response.statusCode, 200);
     const result = response.send.lastCall.args[0];
-    sinon.assert.callCount(getJsonFromApiStub, 1); // Bond was not called to get SA key
-    t.deepEqual({ ...result }, jadeDrsMarthaResult);
+    t.deepEqual(response.body, jadeDrsMarthaResult);
     t.falsy(result.googleServiceAccount);
-    t.is(
-        getJsonFromApiStub.getCall(0).args[0],
-        'https://jade.datarepo-dev.broadinstitute.org/ga4gh/drs/v1/objects/abc',
-    );
-    t.is(getJsonFromApiStub.getCall(0).args[1], 'bearer abc123');
+
+    sinon.assert.callCount(getJsonFromApiStub, 1);
 });
 
 test.serial('martha_v3 parses Gen3 CRDC response correctly', async (t) => {
-    getJsonFromApiStub.onCall(0).resolves(gen3CrdcResponse);
-    getJsonFromApiStub.onCall(1).resolves(googleSAKeyObject);
+    const bond = bondUrls('dcf-fence');
+    const drs = drsUrls(config.HOST_CRDC_STAGING, '206dfaa6-bcf1-4bc9-b2d0-77179f0f48fc');
+    getJsonFromApiStub.withArgs(bond.serviceAccountKeyUrl, terraAuth).resolves(googleSAKeyObject);
+    getJsonFromApiStub.withArgs(drs.objectsUrl, null).resolves(gen3CrdcResponse);
     const response = mockResponse();
+
     await marthaV3(
         mockRequest({ body: { 'url': `dos://${config.HOST_CRDC_STAGING}/206dfaa6-bcf1-4bc9-b2d0-77179f0f48fc` } }),
         response,
     );
+
     t.is(response.statusCode, 200);
-    const result = response.send.lastCall.args[0];
-    sinon.assert.callCount(getJsonFromApiStub, 2);
-    t.deepEqual({ ...result }, gen3CrdcDrsMarthaResult(googleSAKeyObject, null));
-    t.is(
-        getJsonFromApiStub.getCall(0).args[0],
-        `https://${config.HOST_CRDC_STAGING}/ga4gh/drs/v1/objects/206dfaa6-bcf1-4bc9-b2d0-77179f0f48fc`,
-    );
-    t.falsy(getJsonFromApiStub.getCall(0).args[1]); // no auth passed
+    t.deepEqual(response.body, gen3CrdcDrsMarthaResult(googleSAKeyObject, null));
 
-    const requestedBondSAKeyUrl = getJsonFromApiStub.getCall(1).args[0];
-    const saKeyUrlMatches = requestedBondSAKeyUrl.match(bondSAKeyUrlRegEx);
-    t.truthy(saKeyUrlMatches, 'Bond SA key URL called does not match Bond SA key URL regular expression');
-    const expectedSAKeyProvider = 'dcf-fence';
-    const actualSAKeyProvider = saKeyUrlMatches[2];
-    t.is(actualSAKeyProvider, expectedSAKeyProvider);
+    sinon.assert.callCount(getJsonFromApiStub, 2);
 });
 
 test.serial('martha_v3 parses a Gen3 CRDC CIB URI response correctly', async (t) => {
-    getJsonFromApiStub.onCall(0).resolves(gen3CrdcResponse);
-    getJsonFromApiStub.onCall(1).resolves(googleSAKeyObject);
+    const bond = bondUrls('dcf-fence');
+    // TODO: This object ID is inconsistent with gen3CrdcResponse but, for now, it doesn't break
+    // anything. Eventually, when we turn on signed URLs for CRDC, we may want to reconcile the
+    // difference so that we can avoid duplication of these IDs between the tests and test data.
```

Comment:
Note this. I resolved some inconsistencies with BDC tests and test data, but not for other projects. I also didn't comprehensively audit and flag all of the inconsistencies in comments like this, so there are probably more to uncover.

TOL: I like that we have some set of tests that operates against real data that we've captured from real DRS servers. I would love to move the majority of the coverage we have today into more targeted unit tests that can use contrived test data that is more expressive of exactly what Martha behavior is being tested. Those tests should be a lot more self-contained, easier to read, and less prone to accidental breakage.